meson.build

```diff
@@ -43,6 +43,10 @@ if compiler.get_id() == 'gcc'
     '-Wno-narrowing',
     '-Wno-overflow',
     '-flto'])
+  #Treat all warnings as errors if requested
+  if get_option('stoponwarning')
+    compiler_args+=['-Werror','-Wfatal-errors']
```

Comment:
This flag is misnamed I believe.  It means consider errors as fatal, ie stop the build process, which is what we want.
Otherwise gcc continues building despite errors.

Reply:
this is not exactly that, It may go on with the same translation unit, but the build will fail anyway. The point is that adding this flag also removes some information about the failure with may be helpful (for example when it suggests candidate functions). So I'd rather add -fmax-errors=10 if you really wan to prevent hundreds of errors, but this main remain useful.
To me one could add -fmax-errors in any build flavour; adding it with stoponwarning is not the right place IMHO.

Reply:
I see.
But what is the likelihood of having a PR introducing more than one warning ? Isn't it preferable to fail fast ?

Reply:
Really depends on the PR you are doing. Anyway, is you think that PR are really unlikely to introduce more than one warning, why would you ask gcc to stop on the first?
I really don't see why this is useful here... but this is just my opinion.